test/e2e/delete_test.go

```diff
@@ -23,42 +22,46 @@ func testWindowsNodeDeletion(t *testing.T) {
-			e2eMachineSets = append(e2eMachineSets, &machineSet)
-		}
-		if machineSet.Spec.Selector.MatchLabels["machine.openshift.io/os-id"] == "Windows" {
-			windowsMachineSetWithLabel = machineSet.Name
+	var windowsMachineSetWithLabel *mapi.MachineSet
+	for _, machineSet := range e2eMachineSets.Items {
+		if machineSet.Spec.Selector.MatchLabels[clusterinfo.MachineOSIDLabel] == "Windows" {
+			windowsMachineSetWithLabel = &machineSet
+			break
 		}
 	}
-	windowsMachineSet, err := testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Get(context.TODO(),
-		windowsMachineSetWithLabel, meta.GetOptions{})
-	require.NoError(t, err, "error getting Windows MachineSet")
-	// Reset the number of nodes to be deleted to 0
-	gc.numberOfNodes = 0
-	// Delete the Windows VM that got created.
-	windowsMachineSet.Spec.Replicas = &gc.numberOfNodes
-	_, err = testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Update(context.TODO(), windowsMachineSet,
-		meta.UpdateOptions{})
-	require.NoError(t, err, "error updating Windows MachineSet")
-	// we are waiting 10 minutes for all windows machines to get deleted.
-	err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, true, false)
-	require.NoError(t, err, "Windows node deletion failed")
+
+	t.Run("Nodes scale down", func(t *testing.T) {
+		require.NotNilf(t, windowsMachineSetWithLabel, "could not find MachineSet with Windows label")
+		// Set the expected number of Windows nodes to 0
+		gc.numberOfNodes = 0
+
+		// Scale the Windows MachineSet to 0
+		windowsMachineSetWithLabel.Spec.Replicas = &gc.numberOfNodes
+		_, err = testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Update(context.TODO(),
+			windowsMachineSetWithLabel, meta.UpdateOptions{})
+		require.NoError(t, err, "error updating Windows MachineSet")
+
+		// we are waiting 10 minutes for all windows machines to get deleted.
+		err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, true, false)
+		require.NoError(t, err, "Windows node deletion failed")
+	})
 
 	// Cleanup all the MachineSets created by us.
-	for _, machineSet := range e2eMachineSets {
-		assert.NoError(t, testCtx.deleteMachineSet(machineSet), "error deleting MachineSet")
+	for _, machineSet := range e2eMachineSets.Items {
+		assert.NoError(t, testCtx.deleteMachineSet(&machineSet), "error deleting MachineSet")
 	}
-	// Phase is ignored during deletion, in this case we are just waiting for Machines to be deleted.
-	require.NoError(t, testCtx.waitForWindowsMachines(int(gc.numberOfNodes), ""), "Windows machine deletion failed")
 
-	// Test if prometheus configuration is updated to have no node entries in the endpoints object
-	testPrometheus(t)
+	t.Run("Prometheus configuration", func(t *testing.T) {
+		// Phase is ignored during deletion, in this case we are just waiting for Machines to be deleted.
+		require.NoError(t, testCtx.waitForWindowsMachines(int(gc.numberOfNodes), ""), "Windows machine deletion failed")
```

Comment:
This again as nothing to do with the prometheus test, so why is it part of the test? There could be future scenarios where we could decide to skip the test but don't waiting for the machines to be deleted at the call site will result in issues.

Reply:
The connection to the prometheus is test is that Machines being fulled scaled down is a requirement of the test. The test is testing that the windows-exporter service has been updated to remove all the Windows node. 

I can move it out of the test in case of a future scenario like you described

Reply:
So the issue arose from splitting things out in to sub-tests. Having sub-tests are useful and they typically imply that they can be called on their own if needed. Do we really need a sub-test for node scale down that does not check if the Prometheus configuration worked?